src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.tt

```diff
@@ -0,0 +1,135 @@
+
+        public static SupportedSuppressionsBuilder Instance => lazy.Value;
+
+        public ImmutableArray<SuppressionDescriptor> Descriptors { get; }
+
+        private SupportedSuppressionsBuilder()
+        {
+            Descriptors = GetDescriptors();
+        }
+
+        private static ImmutableArray<SuppressionDescriptor> GetDescriptors()
+        {
+            var descriptors = new SuppressionDescriptor[]
+            {
+<#
+                        var idList = GetFilteredDiagnosticIds();
+                        WriteLine("                // Number of diagnostics (C# and VB.NET): " + idList.Count());
+                        foreach(var id in idList)
+                        {
+#>
+                CreateDescriptor("<#= id #>"),
+<#
+                        }
+#>
+            };
+            return ImmutableArray.ToImmutableArray(descriptors);
+        }
+
+        private static SuppressionDescriptor CreateDescriptor(string diagId) =>
+            new SuppressionDescriptor("X" + diagId, diagId, "Suppressed on the Sonar server");
+    }
+}
+<#+
+
+// T4 Text Templates
+// MS docs - see https://docs.microsoft.com/en-us/visualstudio/modeling/code-generation-and-t4-text-templates?view=vs-2022
+//
+// Notes on debugging through the generator
+// ----------------------------------------
+// * change the "debug" value in line 1 to "true"
+// * add a call to "System.Diagnostics.Debugger.Launch();" somewhere in the code below
+// * save this file.
+// A dialogue should pop up asking you attach a debugger.
+//
+//
+
+        // The set of assemblies in which we should look for diagnostics
+        private static readonly Assembly[] AssembliesWithDiagnostics = new Assembly[]
+        {
+            typeof(SonarAnalyzer.Rules.CSharp.CommentKeyword).Assembly,
+            typeof(SonarAnalyzer.Rules.VisualBasic.CommentKeyword).Assembly
+        };
+
+        private static IEnumerable<string> GetAllDiagnosticIds() =>
+            AssembliesWithDiagnostics.SelectMany(asm => asm.GetTypes())
+                                     .SelectMany(type => GetDiagnosticDescriptors(type))
+                                     .Select(diag => diag.Id);
+
+        private static IEnumerable<string> GetFilteredDiagnosticIds()
+        {
+            return GetAllDiagnosticIds()
+                .Distinct()
+                .Where(id => !id.StartsWith("S999"))    // exclude "utility" analyzers
+                .OrderBy(x => x);
+        }
+
+        private static IEnumerable<DiagnosticDescriptor> GetDiagnosticDescriptors(Type t)
+        {
+            try
+            {
+                if (t.GetCustomAttributes(DiagnosticAnalyzerAttribute, false).Any())
+                {
+                    return Create(t).SupportedDiagnostics;
+                }
+
+            }
+            catch(Exception e)
+            {
+                Console.WriteLine(e.Message);
```

Comment:
Where would this be shown? in the build output?

Reply:
Changed to appear in the generated code.